File: rb101_programming_foundations/rb109_general_ruby_assessment/exercises/101-109_review/easy/how_many.rb
# https://launchschool.com/exercises/d5c51b39

# Q: Write a method that counts the number of occurrences of each element in a
# given array

def count_occurrences(arr)
  count = {}
  arr.uniq.each { |x| count[x] = arr.count(x) }
  pp count
end

vehicles = [
  'car', 'car', 'truck', 'car', 'SUV', 'truck',
  'motorcycle', 'motorcycle', 'car', 'truck'
]

count_occurrences(vehicles)

File: exercises/210-javascript_smallproblems/easy_4/how_many.js
// Write a function that counts the number of occurrences of each element in a given array.
// Once counted, log each element alongside the number of occurrences.

const vehicles = ['car', 'car', 'truck', 'car', 'SUV', 'truck',
                'motorcycle', 'motorcycle', 'car', 'truck'];

countOccurrences(vehicles);

function countOccurrences(elements) {
  let count = vehicles.reduce((count, el) => {
    count[el] = (count[el] || 0) + 1;
    return count;
  }, {})

  Object.keys(count).forEach(el => {
    console.log(el + " => " + count[el]);
  });
}